### DISCUSSION TITLE
Error with `npm install`

### DISCUSSION THREAD

**Discussion by gorhill** [ghostery/adblocker#2147] [General]
@remusao, @mjethani

When I git-clone the repo and do a `npm install` in `packages/adblock-benchmark`, I get the following error:

```
> @cliqz/adblocker-benchmarks@1.22.4 prepare [...]/adblocker/packages/adblocker-benchmarks
> yarn download-requests && yarn download-ubo && yarn download-abp && yarn download-tsurlfilter && yarn download-brave && yarn download-fast-hosts-lookup

00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'download-requests'
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @cliqz/adblocker-benchmarks@1.22.4 prepare: `yarn download-requests && yarn download-ubo && yarn download-abp && yarn download-tsurlfilter && yarn download-brave && yarn download-fast-hosts-lookup`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @cliqz/adblocker-benchmarks@1.22.4 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

I can run the benchmarks fine afterward if I run the various `make` targets manually, which I understand are supposed to be run with the command which fails.

I gather this an issue specific to me?

**Comment by mjethani** [comment]
What happens if you run `yarn download-requests` or `npm run download-requests` explicitly?

**Reply by gorhill** [reply]
The first command fails with:

    00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'download-requests'

The second command works with:

```
> @cliqz/adblocker-benchmarks@1.22.4 download-requests [...]/adblocker/packages/adblocker-benchmarks
> make requests.json

curl https://cdn.cliqz.com/adblocking/requests_top500.json.gz | gunzip \
	| sed -e '$ ! s/$/,/; 1 s/^/\[/; $ s/$/]/' > requests.json
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 10.1M  100 10.1M    0     0  2426k      0  0:00:04  0:00:04 --:--:-- 2426k
```

**Reply by mjethani** [reply]
```console
$ yarn -v
1.22.4
```

This seems specific to your setup.

But I also don't know why we have this dependence on [yarn](https://yarnpkg.com/).

**Reply by gorhill** [reply]
> This seems specific to your setup.

Ok thanks, when I enter `yarn -v`, I get an error, it appears `-v` is used for `--verbose` on my side -- and when I try `yarn --version`, I get `0.32+git`. So it seems I have a somewhat incompatible `yarn`.

**Reply by gorhill** [reply]
Actually it seems `yarn` on my side is an entirely different tool than the one expected:

    > man yarn

    YARN(1) General Commands Manual                                                               
  
    NAME
          yarn - scenario testing of Unix command line tools

http://manpages.ubuntu.com/manpages/bionic/man1/yarn.1.html

---

[Which ‘yarn’?](http://www.alvinsim.com/which-yarn/)

**Reply by remusao** [reply]
I think it should work if you run all the commands with `npm`. Yarn is only needed for the workspace mode which is used to manage the different packages of the adblocker repo. The `adblocker-benchmarks` package is somewhat independent so it should not require yarn to function properly.

**Reply by mjethani** [reply]
> I think it should work if you run all the commands with `npm`.

Follow-up: https://github.com/cliqz-oss/adblocker/pull/2148
